### PR TITLE
Add Momentum SGD optimizer implementation

### PR DESCRIPTION
**Pull request by Adhithya-Laxman**
## Description

This PR implements the **Momentum SGD** optimizer using pure NumPy as part of the effort to add **neural_network/optimizers** to the repository.

This PR addresses part of issue #13662 .

## What does this PR do?

- Implements Momentum SGD optimizer that accelerates gradient descent by adding momentum to weight updates
- Uses velocity accumulation to dampen oscillations and speed up convergence
- Provides a clean, educational implementation without external deep learning frameworks

## Implementation Details

- **Algorithm**: SGD with momentum
- **Update rule**: 
  ```
  velocity = momentum * velocity - learning_rate * gradient
  param = param + velocity
  ```
- **Pure NumPy**: No PyTorch, TensorFlow, or other frameworks required
- **Educational focus**: Clear variable names, detailed docstrings, and comments

## Features

✅ Complete docstrings with parameter descriptions  
✅ Type hints for all function parameters and return values  
✅ Doctests for correctness validation  
✅ Usage example demonstrating optimizer on quadratic function  
✅ PEP8 compliant code formatting  
✅ Momentum accumulation with configurable momentum factor  

## Testing

All doctests pass:
```
python -m doctest neural_network/optimizers/momentum_sgd.py -v
```

Linting passes:
```
ruff check neural_network/optimizers/momentum_sgd.py
```

Example output demonstrates proper convergence behavior.

This PR is the second optimizer in the planned sequence outlined in #13662:
- ✅ Stochastic Gradient Descent (SGD) - #13671
- ✅ **Momentum SGD** (this PR)
- ⏳ Nesterov Accelerated Gradient (NAG) - upcoming
- ⏳ Adagrad - upcoming
- ⏳ Adam - upcoming
- ⏳ Muon - upcoming
## References

- [Momentum SGD - Wikipedia](https://en.wikipedia.org/wiki/Stochastic_gradient_descent#Momentum)
- [CS231n: Neural Networks 3](https://cs231n.github.io/neural-networks-3/)

## Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md)
- [x] This pull request is all my own work -- I have not plagiarized
- [x] I know that pull requests will not be merged if they fail the automated tests
- [x] This PR only changes one algorithm file
- [x] All new Python files are placed inside an existing directory
- [x] All filenames are in all lowercase characters with no spaces or dashes
- [x] All functions and variable names follow Python naming conventions
- [x] All function parameters and return values are annotated with Python type hints
- [x] All functions have doctests that pass the automated testing
- [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation
- [x] 

## Next Steps

Additional optimizers (Adam, Adagrad, NAG, Muon) will be submitted in follow-up PRs to maintain focused, reviewable contributions.

### CODE REVIEW

**Comment by Adhithya-Laxman** [comment]
Hi! This PR seems ready for review/merge. Could someone take a look? Thanks!

**Review by cclauss — APPROVED** [review]
